### PR TITLE
feat: move compile project to scheduler

### PR DESCRIPTION
**Pull request by rephus**
Closes: #5165

### Description:

Note: I haven't updated the logic for any of these methods, I'm just moving the async stuff inside scheduler. But these methods are still using the old Job table method , we'll have to refactor the backend and frontend to move this to the new sheduler log details (like csv) . I created separated tickets for this 

Moves both : 
- "Dbt refresh" -> Compile project 
- "Test and compile " (settings) -> TestAndCOmpile project

IMO these 2 functions could be merged together . But I have added a separate ticket for that . 

image

```
2023-05-02 14:12:34 [Lightdash] info: Worker worker-05e44e4151677f91f2 was created
2023-05-02 14:12:34 [Lightdash] debug: Spawned
2023-05-02 14:12:34 [Lightdash] debug: Worker connected and looking for jobs... (task names: 'generateDailyJobs', 'handleScheduledDelivery', 'sendSlackNotification', 'sendEmailNotification', 'downloadCsv', 'compileProject')
2023-05-02 14:12:37 [Lightdash] info: Worker worker-05e44e4151677f91f2 started job 14881 (compileProject). Attempt 1 of 1
2023-05-02 14:12:37 [Lightdash] debug: Found task 14881 (compileProject)
2023-05-02 14:12:37 [Lightdash] http: POST /api/v1/projects/3675b69e-8324-4110-bdca-059031aa8da3/refresh 200 - 37.062 ms

2023-05-02 14:12:37 [Lightdash] debug: Initialize warehouse client of type postgres
...
2023-05-02 14:12:41 [Lightdash] debug: Convert explores
2023-05-02 14:12:41 [Lightdash] debug: Destroy local project adapter
2023-05-02 14:12:41 [Lightdash] debug: Destroy base project adapter
2023-05-02 14:12:42 [Lightdash] http: GET /api/v1/jobs/e9fb9129-2316-4b79-bfe2-d6574675c0e6 200 - 11.299 ms

2023-05-02 14:12:42 [Lightdash] info: Worker worker-05e44e4151677f91f2 successfully executed job 14881 (compileProject)
2023-05-02 14:12:42 [Lightdash] debug: Completed task 14881 (compileProject) with success (4462.50ms)
2023-05-02 14:12:42 [Lightdash] info: Worker worker-05e44e4151677f91f2 concluded job 14881 (compileProject)
2023-05-02 14:12:42 [Lightdash] http: GET /api/v1/projects/3675b69e-8324-4110-bdca-059031aa8da3/explores?filtered=true 200 - 16.745 ms
```

WITH CLI (no test) 
[Screencast from 02-05-23 15:53:55.webm](https://user-images.githubusercontent.com/1983672/235696789-c5235e00-9c19-46fc-9bd5-bc38b7eefbf4.webm)

With Local (test + compile) 
[Screencast from 02-05-23 16:19:19.webm](https://user-images.githubusercontent.com/1983672/235696864-18fdf00f-80ee-46a0-8046-6e75225ef32a.webm)

Screenshot from 2023-05-02 16-19-53

### CODE REVIEW

**Comment by rephus** [comment]
TODO: Remove the old Job ID logic for Compile Project and replace with scheduler log details  (frontend + backend)

**Comment by rephus** [comment]
Created a tech debt ticket for : Unify testAndCompile and CompileProject service methods + scheduler

File: packages/common/src/types/scheduler.ts [hunk 1]
```diff
@@ -234,3 +236,10 @@ export type ApiCsvUrlResponse = {
         status: string;
     };
 };
+
+export type CompileProjectPayload = {
+    userUuid: string;
```
Comment (owlas) [inline packages/common/src/types/scheduler.ts:241]: Is this the user that triggered the payload? Could we give it a name like `createdByUserUuid` if so

File: packages/common/src/types/scheduler.ts [hunk 2]
```diff
@@ -234,3 +236,10 @@ export type ApiCsvUrlResponse = {
...
+    projectUuid: string;
+    requestMethod: string;
```
Comment (owlas) [inline packages/common/src/types/scheduler.ts:243]: There is a `RequestMethod` type we can use here I think (unless graphile doesn't like that)

File: packages/backend/src/services/ProjectService/ProjectService.ts [hunk 3]
```diff
@@ -1108,6 +1161,9 @@ export class ProjectService {
             throw new ForbiddenError();
         }
 
+        // This job is the job model we use to compile projects
+        // This is not the graphile Job id we use on scheduler
+        // TODO: remove this old job method and replace with scheduler log details
```
Comment (owlas) [inline packages/backend/src/services/ProjectService/ProjectService.ts:1166]: How do these work together on this PR? The work is happening on the scheduler but it still updates the old job table?

**Review by owlas — APPROVED** [review]
Nice, so this PR keeps the old jobs table and the old api / method of polling. But the work happens on the scheduler. Great incremental change!

I left a couple of small suggestions, otherwise 🐑 it

File: packages/common/src/types/scheduler.ts (hunk 2 above)
Comment (rephus) [reply packages/common/src/types/scheduler.ts:243]: Yes, I use the method to parse it into a string on the http request, but his is going to write it into a JSON blob in database so I prefer this to be a string when it happens.

File: packages/common/src/types/scheduler.ts (hunk 1 above)
Comment (rephus) [reply packages/common/src/types/scheduler.ts]: Done

**Pull request merged**